File: 0121-best-time-to-buy-and-sell-stock/solution.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        start=prices[0]
        n=len(prices)
        ans=0
        temp=0
        for i in range(1,n):
            if start>prices[i]:
                start=prices[i]
            else:
                temp=prices[i]-start
                if temp>ans:
                    ans=temp
        return ans

File: 0121-best-time-to-buy-and-sell-stock/solution.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int curr_min=prices[0];
        int res=0;
        for(int price:prices) {
            curr_min=min(curr_min,price);
            res=max(res,price-curr_min);
        }
        return res;
    }
};
